### PR TITLE
Added Base DbContext

### PR DESCRIPTION
**Pull request by SejalPatel29**

### CODE REVIEW

File: Scrips.BaseDbContext/Scrips.BaseDbContext/ConnectionString.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace Scrips.BaseDbContext
+{
+    public sealed class ConnectionString
+    {
+        public ConnectionString(IConfiguration configuration)
+        {
+            DefaultConnection = configuration.GetConnectionString("SchedulingDb");
```
Comment (tariqalardah) [inline Scrips.BaseDbContext/Scrips.BaseDbContext/ConnectionString.cs]: I think "schedulingDb" here by mistake, do you mean LoggingDb?

File: Scrips.BaseDbContext/Scrips.BaseDbContext/ApplicationDbContext.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Scrips.BaseDbContext.Dtos;
+using Scrips.BaseDbContext.Entities;
+using Serilog;
+using System.Data;
+
+namespace Scrips.BaseDbContext
+{
+    public class ApplicationDbContext : DbContext
```
Comment (tariqalardah) [inline Scrips.BaseDbContext/Scrips.BaseDbContext/ApplicationDbContext.cs]: Better to have a more descriptive name like BaseDbContext or AuditableBaseDbContext

**Review by tariqalardah — COMMENTED** [review]
Done with 2 minor comments

File: Scrips.BaseDbContext/Scrips.BaseDbContext/ConnectionString.cs (hunk 1 above)
Comment (SejalPatel29) [reply Scrips.BaseDbContext/Scrips.BaseDbContext/ConnectionString.cs]: Class is no longer in use so i removed it.

File: Scrips.BaseDbContext/Scrips.BaseDbContext/ApplicationDbContext.cs (hunk 2 above)
Comment (SejalPatel29) [reply Scrips.BaseDbContext/Scrips.BaseDbContext/ApplicationDbContext.cs]: Change to AuditableBaseDbContext

**Review by tariqalardah — APPROVED** [review]

**Review by tariqalardah — CHANGES_REQUESTED** [review]
The code is logging the real value anyway
Please check the code snippet on confluence page

File: Scrips.BaseDbContext/Scrips.BaseDbContext/AuditableBaseDbContext.cs [hunk 3]
```diff
@@ -84,28 +86,31 @@ private List<LogAudit> DetectChanges()
...
                                 entry.AuditActionType = AuditActionType.Deleted;
-                                if (prop.Metadata.PropertyInfo.GetCustomAttribute<AuditIngore>() is null)
-                                    entry.OldValues[prop.Metadata.Name] = prop.OriginalValue;
+                                entry.OldValues[prop.Metadata.Name] = maskValue ? maskedValue : prop.OriginalValue;
                                 break;
                             case EntityState.Added:
                                 entry.AuditActionType = AuditActionType.Added;
-                                if (prop.Metadata.PropertyInfo.GetCustomAttribute<AuditIngore>() is null)
-                                    entry.NewValues[prop.Metadata.Name] = prop.CurrentValue;
+                                if (prop.Metadata.PropertyInfo.GetCustomAttribute<MaskValueAuditAttribute>() is null)
```
Comment (tariqalardah) [inline Scrips.BaseDbContext/Scrips.BaseDbContext/AuditableBaseDbContext.cs]: if is not needed here!
it is already covered by (maskValue ?)

File: Scrips.BaseDbContext/Scrips.BaseDbContext/AuditableBaseDbContext.cs [hunk 4]
```diff
@@ -84,28 +86,31 @@ private List<LogAudit> DetectChanges()
...
+                                    entry.NewValues[prop.Metadata.Name] = maskValue ? maskedValue : prop.CurrentValue;
                                 break;
                             case EntityState.Modified:
                                 if (prop.IsModified)
                                 {
                                     entry.AuditActionType = AuditActionType.Updated;
-                                    if (prop.Metadata.PropertyInfo.GetCustomAttribute<AuditIngore>() is null)
+                                    if (prop.Metadata.PropertyInfo.GetCustomAttribute<MaskValueAuditAttribute>() is null)
```
Comment (tariqalardah) [inline Scrips.BaseDbContext/Scrips.BaseDbContext/AuditableBaseDbContext.cs]: if is not needed here!
it is already covered by (maskValue ?)

File: Scrips.BaseDbContext/Scrips.BaseDbContext/MaskValueAuditAttribute.cs [hunk 5]
```diff
@@ -6,9 +6,9 @@
 
 namespace Scrips.BaseDbContext
 {
-    public class AuditIngore : Attribute
+    public class MaskValueAuditAttribute : Attribute
```
Comment (tariqalardah) [inline Scrips.BaseDbContext/Scrips.BaseDbContext/MaskValueAuditAttribute.cs:13]: Add the documentation found on confluence

**Review by tariqalardah — CHANGES_REQUESTED** [review]
Still need review

File: Scrips.BaseDbContext/Scrips.BaseDbContext/AuditableBaseDbContext.cs (hunk 4 above)
Comment (SejalPatel29) [reply Scrips.BaseDbContext/Scrips.BaseDbContext/AuditableBaseDbContext.cs]: Sorry, I forgot to remove it.

File: Scrips.BaseDbContext/Scrips.BaseDbContext/MaskValueAuditAttribute.cs (hunk 5 above)
Comment (SejalPatel29) [reply Scrips.BaseDbContext/Scrips.BaseDbContext/MaskValueAuditAttribute.cs:13]: okay

**Review by tariqalardah — APPROVED** [review]

**Pull request merged**